workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityWorkerMenuGenerator.tt

```diff
@@ -62,21 +62,85 @@ var workerType = workerTypeList[i];
             WorkerBuilder.Clean();
             Debug.Log("Clean completed");
         }
-        
-        private static void MenuBuild(BuildEnvironment environment, params string[] platforms)
+
+        private static bool CheckWorkersCanBuildForEnvironment(BuildEnvironment environment, string[] workerTypes)
+        {
+            var canInitiateBuild = true;
+            var spatialOSBuildConfiguration = SpatialOSBuildConfiguration.GetInstance();
+
+            foreach (var workerType in workerTypes)
+            {
+                var buildTargetsForWorker = WorkerBuilder.GetBuildTargetsForWorkerForEnvironment(workerType, environment);
+                var buildTargetsMissingBuildSupport = BuildSupportChecker.GetBuildTargetsMissingBuildSupport(buildTargetsForWorker);
+
+                if (buildTargetsMissingBuildSupport.Length > 0)
+                {
+                    canInitiateBuild = false;
+
+                    Debug.LogError(BuildSupportChecker.ConstructMissingSupportMessage(workerType, environment, buildTargetsMissingBuildSupport),
+                        spatialOSBuildConfiguration);
+                }
+            }
+
+            return canInitiateBuild;
+        }
+
+        private static void MenuBuild(BuildEnvironment environment, params string[] workerTypes)
         {
             // Delaying build by a frame to ensure the editor has re-rendered the UI to avoid odd glitches.
             EditorApplication.delayCall += () =>
             {
-                LocalLaunch.BuildConfig();
+                if (!CheckWorkersCanBuildForEnvironment(environment, workerTypes))
+                {
+                    DisplayBuildFailureDialog();
```

Comment:
This doesn't look like it will tell them why it failed.

Reply:
Actually isn't this checked in the build script itself and an error would be given there?

Reply:
It tells them to check the Unity Console Window which should have the information as `Check` prints errors when it will return false